superset/views/base.py

```diff
@@ -256,10 +256,13 @@ def json_response(
 
 
 def menu_data() -> Dict[str, Any]:
-    menu = appbuilder.menu.get_data()
+    menu = []
     root_path = "#"
     logo_target_path = ""
-    if not g.user.is_anonymous:
+    user_is_anonymous = True
+    if hasattr(g, "user") and not g.user.is_anonymous:
```

Comment:
This defeats the `Public` role. `menu.get_data` is already checking if the user is authenticated and if not getting permissions from the `Public` role. Is there any other blocker?

Reply:
I think the main issue is that the public role doesn't work. I was getting an error on the sign in view for `menu.get_data`.

Also this test https://github.com/preset-io/incubator-superset/blob/master/tests/schedules_test.py#L495 was failing without the `hasattr(g, "user")` check.

Reply:
Isn't that a flaky test? would risk to say it was a coincidence. 
Re checked and it works with anonymous and the public role, https://github.com/dpgaspar/Flask-AppBuilder/blob/master/flask_appbuilder/tests/test_menu.py#L101 but we need to add the endpoint to the public role. Yet `menu.get_data` shoud work